lua/autorun/server/sv_daily_restart.lua

```diff
@@ -1,67 +1,88 @@
-if SERVER then
-    util.AddNetworkString("AlertUsersOfRestart")
-    local desiredRestartTime = 3    -- The hour to initiate a restart. Must be between 0-24
-    local warningTime = 15          -- How many minutes players have until restart...
-    local restartAt                 -- Time at which the server will restart
-
-    function restartServer()
-        print("restarting server")
-        -- TODO: Write the latest restart time to a local file
-        --local params = {}
-        --local headers = {}
-        --http.Post("localhost:2327/gmod/restart", params, function() end, function() end, headers)
-    end
+util.AddNetworkString("AlertUsersOfRestart")
 
-    function sendAlertToClients(message)
-        for k, v in pairs(player.GetHumans()) do
-            v:ChatPrint(message)
-        end
-    end
+local CFCDailyRestart = {}
+CFCDailyRestart.desiredRestartTime      = 3                         -- The hour to initiate a restart. Must be between 0-24
+CFCDailyRestart.warningTime             = 45                        -- How many minutes players have until restart...
+CFCDailyRestart.restartAt               = nil                       -- Time at which the server will restart
+CFCDailyRestart.fileName                = "CFC_LastDailyRestart"
+CFCDailyRestart.countingDownToRestart   = false
+CFCDailyRestart.lastRestart             = file.Read(fileName..".txt", "DATA")
 
-    function alertClientsOfServerRestart()
-        print("Alerting clients of Restart")
-        restartAt = os.time() + (warningTime * 60)
+local function CFCDailyRestart:restartServer()
+    print("restarting server")
+    
+    file.Write(fileName..".txt", os.time())
 
-        -- TODO: Alert how many seconds left
-        -- You'll want a logarithmic-ish function to alert at:
-        -- 15 min, 10min, 5min, 4min, 3min,2min,1min,30s,15s,10,9,8.. so on
-        sendAlertToClients()
+    --local params = {}
+    --local headers = {}
+    --http.Post("localhost:2327/gmod/restart", params, function() end, function() end, headers)
+end
 
-        net.Start("AlertUsersOfRestart")
-            net.WriteFloat(restartAt)
-        net.Broadcast()
+local function CFCDailyRestart:sendAlertToClients(message)
+    for k, v in pairs(player.GetHumans()) do
+        v:ChatPrint("[CFC Daily Restart] "..message)
     end
+end
+
+local function CFCDailyRestart:alertClientsOfServerRestart()
+    print("Alerting clients of Restart")
+    CFCDailyRestart.restartAt = os.time() + (warningTime * 60)
+
+    CFCDailyRestart:sendAlertToClients("Restarting server in "..CFCDailyRestart.warningTime.." minutes.")
+
+    net.Start("AlertUsersOfRestart")
+        net.WriteFloat(restartAt)
+    net.Broadcast()
+end
 
-    --TODO: Create a timer to 
+timer.Create("CFC_DailyRestart", 1, 0, function()
```

Comment:
This function got pretty big. Can we take it out of the `timer.Create and give it its own name?

Let's try to break this up into a couple of other functions, too. It's a little longer that I'd like.

A good thing to keep in mind is that your function should try to do only **1** thing.